tests/dogtag/pytest-ansible/pytest/performance_test/test_acme_cert_enrollment.py

```diff
@@ -0,0 +1,238 @@
+              cmd_input=protected_input, err_msg="OpenSSL Error")
+    data = json.dumps({"protected": protected64, "payload": payload64, "signature": b64(out)})
+    return do_request(url, data=data.encode('utf8'), err_msg=err_msg)
+
+
+# helper function - poll until complete
+def poll_until_not(account_key, url, statuses, err_msg, acct_headers=None, nonce=None):
+    result, t0 = None, time.time()
+    while result is None or result['status'] in statuses:
+        assert (time.time() - t0 < 3600), "Polling timeout"  # 1 hour timeout
+        time.sleep(0 if result is None else 2)
+        result, _, _, nonce = send_signed_request(account_key, url, None, err_msg, acct_headers=acct_headers,
+                                                  nonce=nonce)
+    return result, nonce
+
+
+def get_crt(domain_name, acme_dir, log=LOGGER, nonce=None):
+    # create a new order
+    log.info("Creating new order...")
+    order_payload = {"identifiers": [{"type": "dns", "value": d} for d in {domain_name}]}
+    order, _, order_headers, nonce = send_signed_request(account_key, directory['newOrder'], order_payload,
+                                                         "Error creating new order", acct_headers=acct_headers,
+                                                         nonce=nonce)
+    log.info("Order created!")
+
+    # get the authorizations that need to be completed
+    for auth_url in order['authorizations']:
+        authorization, _, _, nonce = send_signed_request(account_key, auth_url, None, "Error getting challenges",
+                                                         acct_headers=acct_headers, nonce=nonce)
+        domain = authorization['identifier']['value']
+        log.info("Verifying {0}...".format(domain))
+
+        # find the http-01 challenge and write the challenge file
+        challenge = [c for c in authorization['challenges'] if c['type'] == "http-01"][0]
+        token = re.sub(r"[^A-Za-z0-9_\-]", "_", challenge['token'])
+        keyauthorization = "{0}.{1}".format(token, thumbprint)
+        wellknown_path = os.path.join(acme_dir, token)
+        with open(wellknown_path, "w") as wellknown_file:
+            wellknown_file.write(keyauthorization)
+
+        # say the challenge is done
+        authorization, _, _, nonce = send_signed_request(account_key, challenge['url'], {},
+                                                         "Error submitting challenges: {0}".format(domain),
+                                                         acct_headers=acct_headers, nonce=nonce)
+    authorization, nonce = poll_until_not(account_key, auth_url, ['ready', 'invalid'],
+                                          "Error checking challenge status for {0}".format(domain),
+                                          acct_headers=acct_headers, nonce=nonce)
+    if authorization['status'] != "valid":
+        raise ValueError("Challenge did not pass for {0}: {1}".format(domain, authorization))
```

Comment:
I think we want to wait for the order status to become either ready or invalid, i.e. not pending, so the code probably should look like this:
```
order, nonce = poll_until_not(account_key, order_headers['Location'], ['pending'], ...)
if order['status'] != 'ready':
    raise ...
```